controller/cmd/service-mirror/main.go

```diff
@@ -36,12 +40,18 @@ func Main(args []string) {
 		log.Fatalf("Failed to initialize K8s API: %s", err)
 	}
 
+	probeEvents := make(chan interface{}, probeChanBufferSize)
```

Comment:
how easy is it to expose queue depth as a prometheus metric?  how can we tell if this queue is backing up?

Reply:
a simple approach would be to expose the queue length `len(probeEvents)` as a gauge.

a more sophisticated approach would be to have counters of enqueues and dequeues.  The counter approach has the benefit that not only can we see queue depth by subtrating dequeues from enqueues, but we can also have a label for event type to see breakdowns by event type.